src/routing_table/mod.rs

```diff
@@ -968,78 +912,67 @@ impl RoutingTable {
-        if self.our_version != version {
+        if self.our_prefix.version() != version {
             debug!(
-                "{:?} Not splitting our section with {:?} ver. {}, \
+                "{:?} Not splitting our section with {:?}, \
                  update is for a different version: {}",
                 self.our_name,
                 self.our_prefix,
-                self.our_version,
                 version
             );
             return Vec::new(); // Wrong version.
         }
         let next_bit = self.our_name.bit(self.our_prefix.bit_count());
-        let other_prefix = self.our_prefix.pushed(!next_bit);
-        self.our_prefix = self.our_prefix.pushed(next_bit);
+        let other_prefix = self.our_prefix.pushed(!next_bit).with_incremented_version();
+        self.our_prefix = self.our_prefix.pushed(next_bit).with_incremented_version();
         let (our_new_section, other_section) =
             self.our_section.iter().partition::<BTreeSet<_>, _>(
                 |name| self.our_prefix.matches(name),
             );
         self.our_section = our_new_section;
-        self.our_version = version + 1;
         // Drop sections that ceased to be our neighbours.
         let sections_to_remove = self.sections
             .keys()
             .filter(|prefix| !prefix.is_neighbour(&self.our_prefix))
             .cloned()
             .collect_vec();
-        self.insert_new_section(other_prefix, version + 1, other_section);
+        self.insert_new_section(other_prefix, other_section);
         sections_to_remove
             .into_iter()
-            .filter_map(|prefix| {
-                self.sections.remove(&prefix).map(|(_, section)| section)
-            })
+            .filter_map(|prefix| self.sections.remove(&prefix))
             .flat_map(BTreeSet::into_iter)
             .collect()
     }
 
     /// Inserts the given section. Logs an error if it already exists.
-    fn insert_new_section(&mut self, prefix: Prefix, version: u64, section: BTreeSet<XorName>) {
+    fn insert_new_section(&mut self, prefix: Prefix, section: BTreeSet<XorName>) {
         match self.sections.entry(prefix) {
             Entry::Vacant(entry) => {
-                let _section_ref = entry.insert((version, section));
+                let _section_ref = entry.insert(section);
             }
-            Entry::Occupied(entry) => {
+            Entry::Occupied(mut entry) => {
                 error!(
                     "{:?} Inserting section {:?}, but already has members {:?}. This is a bug!",
                     self.our_name,
                     prefix,
                     entry.get()
                 );
-                let &mut (ref mut v, ref mut s) = entry.into_mut();
-                if *v > version {
-                    return; // Wrong version.
-                }
-                *v = version;
-                s.extend(section);
+                entry.get_mut().extend(section);
```

Comment:
do we need to carry out version check before extend?

Reply:
Possibly. I guess this change might miss the case where an entry with the same prefix bits, but different version exists. Let me look into it...